### PR TITLE
KAFKA-16327: Removed Deprecated variable StreamsConfig#TOPOLOGY_OPTIMIZATION

### PR DESCRIPTION
**Pull request by abhi-ksolves**
Deprecated Variable StreamsConfig#TOPOLOGY_OPTIMIZATION is removed

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -846,14 +846,6 @@ public class StreamsConfig extends AbstractConfig {
...
             "If greater or equal to 0, the summary log will be output according to the set time interval;\n" +
             "If less than 0, summary output is disabled.";
-    /**
-     * {@code topology.optimization}
-     * @deprecated since 2.7; use {@link #TOPOLOGY_OPTIMIZATION_CONFIG} instead
-     */
-    @Deprecated
-    public static final String TOPOLOGY_OPTIMIZATION = TOPOLOGY_OPTIMIZATION_CONFIG;
-
-
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:828]: Nit: we should not remove all blank lines, but leave exactly one blank line

**Review by mjsax — COMMENTED** [review]
Overall LGTM. Thanks for the PR. One minor nit.

**Comment by mjsax** [comment]
I just checked the full repo, and also found `TOPOLOGY_OPTIMIZATION` mentioned in `docs/streams/upgrade_guide.html` -- I think we should update it there, too.

**Comment by abhi-ksolves** [comment]
> I just checked the full repo, and also found `TOPOLOGY_OPTIMIZATION` mentioned in `docs/streams/upgrade_guide.html` -- I think we should update it there, too.

Yes, Mentioned changes are updated

**Comment by abhi-ksolves** [comment]
Hi @mjsax,
Could you please review this PR? I attempted to rebase it, but now it contains too many commits. Should I close this PR and create a new one?

**Comment by abhi-ksolves** [comment]
Hi @mjsax can you please check & confirm?

**Comment by mjsax** [comment]
Not sure what happened to this PR, but it contains 45 commits and thoughts of LOC... Can you please cleanup the PR by rebasing it to latest `trunk` such that it only contains the relevant changes? And we can take it from there.

**Comment by abhi-ksolves** [comment]
Hi @mjsax these commits come from while doing rebasing itself. I will make new PR.

**Comment by mjsax** [comment]
Seems there is conflicts. Can you try again to rebase? (Did you create the new branch from latest `trunk`? Wondering why there are still conflicts...)

**Comment by abhi-ksolves** [comment]
@mjsax Conflicts resolved. Please review again

**Comment by mjsax** [comment]
Thanks for the PR! Merged to `trunk`.

**Pull request merged**